### DISCUSSION TITLE
Exposing additional hashes of public keys

### DISCUSSION THREAD

**Discussion by livenson** [isambard-sc/clifton#13] [General]
Hi,

noticed this - https://github.com/isambard-sc/clifton/blob/d78005504738b327f78aa373a3a281d69d79d2e6/src/main.rs#L133

We can expose additional hashes, md5 probably will stay even though it is not super secure, to support some legacy docs/tools, but we can add more. What algorithm is a preferred one?

**Comment by milliams** [comment]
If I run `ssh-keygen -lf ~/.ssh/id_ed25519` it defaults to SHA256:
```
256 SHA256:P+slihgZ9B4Iv/mSu0E3lbiLFVhyZZsYE/C7JYLBas0 matt@HEX (ED25519)
```
It's possible with that to pass `-E md5` to explicitly ask for the MD5 hash, but the default has moved to SHA256. It seems that OpenSSH have deprecated MD5 and it will be harder to use in the future. For example, the Rust library we use [doesn't even implement it](https://docs.rs/ssh-key/latest/ssh_key/enum.Fingerprint.html).

I'd probably add in support for both SHA256 and possibly SHA512. I think `cryptography` support both of these. When doing a lookup in Waldur, we can always scan the different columns to find a match if needed.

**Comment by livenson** [comment]
ok, thanks, I've added a dev task for that, backref is WAL-7137. Hope to get it in in a week or so.

**Comment by livenson** [comment]
just a heads up - latest develop branch of mastermind contains additional hashes, sha256 and sha512
